_posts/2022-10-09-decentralization-of-zk-rollups.md

```diff
@@ -0,0 +1,192 @@
+
+
+
+1. Charge high fees in order to include transactions in the block. 
+2. Prevent withdrawals. If they have a short position on the network token of the rollup, this could give them economic rewards. 
+3. Lock up supplies of certain tokens. This allows them to reap economic benefits from liquidity pools of said tokens. 
+
+In addition to these DOS attacks, there is also a Slashing attack to consider. This is possible because of the slashing condition that punishes stakers who do not create a block within a specified time X. If a block producer refuses to include transactions from other block producers, even with blocks containing their own transactions, in favor of uncling them the stakers will get slashed. If the block producer conducting this attack has a combined active stake with at least a ⅓ of active voting power, then this could be an effective, with-standing attack. Otherwise, it would be fairly rare for a small staker to get two blocks in a row, so it would not have much effect in the long-run. 
+
+
+## Proof of Efficiency
+
+Proof of Efficiency is a new consensus mechanism put forward to solve the criticisms of PoS on rollups[^3]. It involves a two-step model that splits activities between a sequencer and an aggregator. 
+
+A sequencer is responsible for pre-processing and posting L2 transactions to L1 by submitting a transaction that has the details of the L2 transactions in the calldata field. The sequencer has to pay the L1 gas fee as well as an additional fee in the network token to discourage bad actors. This usage of calldata has certain storage limitations to be considered as well. Even with the intense amount of optimizations it has been through it can only store a limited number of transactions. The average Ethereum block uses about 80KB of data, and with the new EIP-4844 proposal, the calldata (blob) field will provide on average 1MB of data.[^4] This means the base layer could theoretically support a 10X in throughput from rollups. This is discounting compression techniques that could potentially give space savings of up to 70%. There is also a misalignment with this reward and cost structure of the sequencer, as they get charged for the cost of posting the calldata but get rewarded by the L2 gas their transactions generate, which in turn encourages sequencers to favor gas-heavy transactions over simple transactions. During times of network congestion this means that simple transactions may become deprioritized and postponed for an indefinite time, no matter its importance to the user. A solution to this would be to implement a fee structure similar to EIP-1559[^5]<sup>,</sup>[^6], where there is a base fee[^7] and tip. The tip would represent the importance of the transaction to the user and would be the reward for the sequencer. This allows a prioritization of transactions based on importance to the user, rather than a crude one of gas.
```

Comment:
Need references for `This is discounting compression techniques that could potentially give space savings of up to 70%.`, and need another word for `discounting`. (not including?)

Reply:
Understood. Will add.